public/app/features/explore/correlationEditLogic.test.ts

```diff
@@ -0,0 +1,39 @@
+import { CORRELATION_EDITOR_POST_CONFIRM_ACTION } from 'app/types';
+
+import { showModalMessage } from './correlationEditLogic';
+
+// note, closing the editor does not care if isLeft is true or not. Both are covered for regression purposes.
+describe('correlationEditLogic', function () {
+  it.each`
+    action                                                      | isLeft   | dirCor   | dirQuer  | expected
+    ${CORRELATION_EDITOR_POST_CONFIRM_ACTION.CLOSE_PANE}        | ${false} | ${false} | ${false} | ${undefined}
+    ${CORRELATION_EDITOR_POST_CONFIRM_ACTION.CLOSE_PANE}        | ${false} | ${true}  | ${false} | ${'Closing the pane will cause the correlation in progress to be lost. Would you like to save before continuing?'}
```

Comment:
Changing the label doesn't mark a correlation as dirty. Is it expected?


https://github.com/grafana/grafana/assets/745532/796bb9a7-ddbb-4f14-8667-4dd7e01aef24

Reply:
No - changing the label means the correlation dirty state is true. This was a regression when I used the default value as a default value instead of the default value being blank and only using the default value on save. 

And actually I think I'll be able to tackle this and the mixed datasource problem at the same time, so I'll be looking into this more.